### ISSUE TITLE
Validate `--github-api-url`

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#1980]
## Problem
```
infracost comment github --help

      --github-api-url string     GitHub API URL (default "https://api.github.com")
```

When implementing the GitHub comment feature, we struggled with knowing why it wasn’t working. Once we changed the comment mode from “update” to “new” it logged the actual URL it was trying to post to. We use GitHub Enterprise and provided an endpoint that included the “/api/v3/” suffix which Infracost appends on its own. The logging is inconsistent between these functions, and it should probably attempt to detect whether the suffix already exists before appending it.

## Proposed solution
Since we [create github API clients](https://github.com/infracost/infracost/blob/master/internal/comment/github.go#L80-L122), should we check the `--github-api-url` and log a user-friendly error if the user has already added `/api/v3`?

**Comment by varshneydevansh** [comment]
Can you assign this to me?

**Comment by alikhajeh1** [comment]
Thanks @varshneydevansh !

**Comment by varshneydevansh** [comment]
Okay, so from yesterday I set up the project and learned about how it's working with Terraform. (Henceforth, I have also gained some knowledge about it.)

So, according to the above proposed solution, this is what I came up with in the [internal/comment/github.go](https://github.com/infracost/infracost/blob/master/internal/comment/github.go#L80-L122)

```go
// Check if "/api/v3/" already exists in the URL path.
if strings.Contains(u.Path, "/api/v3/") {
	return nil, nil, fmt.Errorf("The /api/v3/ suffix should not be included in the --github-api-url.")
}
```
But, I think if the user is already aware about the /api/v3.
Instead of giving an error, let the user keep going if they put /api/v3 in the apiURL. This would be a good experience because the user already knows about it.

Somewhat like this?
```go
// If "/api/v3/" already exists, do not append "/api/".
if strings.Contains(u.Path, "/api/v3/") {
    apiURL = u.String()
} else {
	// Add api to path if it doesn't exist
	if !strings.HasSuffix(u.Path, "/api/") {
		u.Path += "api/"
	}
	apiURL = u.String()
}
```

Guide me if I am missing anything here.


For the testing, I have downloaded the required file from the https://www.infracost.io/docs/#2-get-api-key
I will do it today.

**Comment by alikhajeh1** [comment]
Thanks! Given that the user might add env vars for this variable in CI, it's probably best if we ensure they set the correct URL as we might use it in other places too.

I'd just update https://github.com/infracost/infracost/blob/master/internal/comment/github.go#L116 to add this check

```
if strings.HasSuffix(u.Path, "/api/v3/") {
  return nil, nil, errors.Wrap(err, "API URL should be the base URL without any paths, remove the /api/v3 suffix")
}
```

**Comment by varshneydevansh** [comment]
Ohh. That makes sense that it could be used in other places too. So, should I do the testing for this, or can I just create a PR?
As this could be my first code PR which will be approved, although my first code PR is still pending for approval in Phoenix Framework project.

I am getting this in my repo after committing. 
```git
Able to merge. These branches can be automatically merged.
```

Regards.

**Comment by aliscott** [comment]
@RomeLeader this is now released in v0.10.25. Thanks @varshneydevansh for your contribution!
